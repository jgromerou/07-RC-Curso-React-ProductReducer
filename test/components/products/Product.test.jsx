import { render, screen } from '@testing-library/react';
import { Product } from '../../../src/components/products/Product';
import { dataProducts } from '../../fixtures/products';

describe('TEST - COMPONENT <Product />', () => {
  const data = dataProducts;
  test('CONTROL - Validar la estructura del componente', () => {
    const snapshot = render(<Product products={data} />);
    expect(snapshot).toMatchSnapshot();
    //screen.debug();
  });

  test('CONTROL -  Validar las props data recibidas del componente', () => {
    render(<Product products={data} />);
    expect(screen.getByText(data[0].title));
    //expect(screen.getByText(data[0].title));
  });

  test('CONTROL - Validar el tag src de la imagen tenga informaciÃ³n', () => {
    render(<Product products={data} />);
    const img = screen.getByRole('img').src;
    const price = screen.getByRole('heading', { name: '$' + data[0].price });

    expect(img).toContain(data[0].image);
    expect(price.textContent).toContain('$' + data[0].price);
    //screen.debug();
  });
});
