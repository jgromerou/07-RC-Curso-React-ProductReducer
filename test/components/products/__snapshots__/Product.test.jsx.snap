// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST - COMPONENT <Product /> CONTROL - Validar la estructura del componente 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="col-sm-6 col-md-4 col-lg-3 mt-2"
      >
        <ul
          class="list-group shadow"
        >
          <li
            class="list-group-item"
          >
            <div
              class="media align-items-lg-center flex-column flex-lg-row p-3"
            >
              <div
                class="media-body order-2 order-lg-1"
              >
                <h5
                  class="mt-0 font-weight-bold mb-2"
                >
                  SAMSUNG A13
                </h5>
                <p
                  class="font-italic text-muted mb-0 small"
                >
                  Nuevo dispositivo con dos procesadores y camara de 13 mpx
                </p>
                <div
                  class="d-flex align-items-center justify-content-between mt-1"
                >
                  <h6
                    class="font-weight-bold my-2"
                  >
                    $
                    120.000
                  </h6>
                </div>
              </div>
              <img
                alt="Generic placeholder image"
                class="ml-lg-5 order-1 order-lg-2"
                src="https://i.imgur.com/KFojDGa.jpg"
                width="200"
              />
              <div
                class="mt-4"
              >
                <button
                  class="btn btn-outline-primary"
                  type="button"
                >
                  MeGusta: 0
                </button>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="col-sm-6 col-md-4 col-lg-3 mt-2"
    >
      <ul
        class="list-group shadow"
      >
        <li
          class="list-group-item"
        >
          <div
            class="media align-items-lg-center flex-column flex-lg-row p-3"
          >
            <div
              class="media-body order-2 order-lg-1"
            >
              <h5
                class="mt-0 font-weight-bold mb-2"
              >
                SAMSUNG A13
              </h5>
              <p
                class="font-italic text-muted mb-0 small"
              >
                Nuevo dispositivo con dos procesadores y camara de 13 mpx
              </p>
              <div
                class="d-flex align-items-center justify-content-between mt-1"
              >
                <h6
                  class="font-weight-bold my-2"
                >
                  $
                  120.000
                </h6>
              </div>
            </div>
            <img
              alt="Generic placeholder image"
              class="ml-lg-5 order-1 order-lg-2"
              src="https://i.imgur.com/KFojDGa.jpg"
              width="200"
            />
            <div
              class="mt-4"
            >
              <button
                class="btn btn-outline-primary"
                type="button"
              >
                MeGusta: 0
              </button>
            </div>
          </div>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
